### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?
Some maintenance work

- Adds **codeclimate** configuration locally. It still needs to be further tuned but now it can be done in place
   - Adjusted config for python code (activated bandit plugin for security checks)
   - DISABLED javascript checks (pending @ignapas and @odeimaiz to setup )
-  ``scripts/code-climate.sh`` to run code-climate in place
```cmd
   ./scripts/code-climate.sh analyze services/web/server/src  
```
- **Makefile**:  new target to run code-climate in place.
```cmd
make code-analysis
# validates config .codeclimate.yml 
...
# Runs analysis
```
- **Makefile**: Fixes issue of spaces in help recipe
- **Makefile**: Refactored recipes for ``build*`` targets. Reduced code redundancy and extended combinations. Some examples (Remove dry-run option (``-n``) to actually run the recipe)
```cmd
 make -n build-devel-nc
 make -n build-nc target="storage webserver"
 make -n build target=storage
 make -n build info-images
...
```
- **Makefile**: added new target ``devenv-all`` to install extra dev tools E.g. builds ``script`` tooling, pulls qx-compile image, etc
```
$  make help
...
devenv               create a python virtual environment with dev tools (e.g. linters, etc)
devenv-all           sets up extra development tools (everything else besides python)
...
```
- Removes **deprecated**:  ``ops`` and ``ancillary`` directories. All ops functionality moved to ITISFoundation/osparc-ops repo
- **Dockerfiles** adds PYTHONOPTIMIZED in for production
- Fixes issues detected by **code analysis** in **webserver** (and a bit in others):
   - bandit ``Use of assert detected``: checks whether assert is
   - Removed deprecated TODOs and FIXMEs
   - Removed deprecated modules (eg. ``s3*``)

## How to test

Try makefile commands above.

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,9 +55,9 @@ export DOCKER_REGISTRY  ?= itisfoundation
 .PHONY: help
 help: ## help on rule's targets
 ifeq ($(IS_WIN),)
-	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+	@awk --posix 'BEGIN {FS = ":.*?## "} /^[[:alpha:][:space:]_-]+:.*?## / {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
```
Comment (sanderegg) [inline Makefile]: long live awk ;) welcome in the 2000s 🤣

File: Makefile [hunk 2]
```diff
@@ -68,58 +69,51 @@ endif
...
-	docker-compose -f services/docker-compose-build.yml build --parallel --no-cache $(target)
+	$(_docker_compose_build) $(target)
 endif
 
-.PHONY: build-devel
-build-devel: .env ## Builds development images and tags them as 'local/{service-name}:development'. For single target e.g. 'make target=webserver build-devel'
-	# Compiling front-end
-	@$(MAKE) -C services/web/client touch compile-dev
+
+build-devel build-devel-nc: .env ## Builds development images and tags them as 'local/{service-name}:development'. For single target e.g. 'make target=webserver build-devel'
```
Comment (sanderegg) [inline Makefile]: nice!

File: Makefile (hunk 1 above)
Comment (mguidon) [reply Makefile]: Be careful. I got a PhD using sed, awk, vi and cvs.

Comment (sanderegg) [reply Makefile]: well he was using awk from 1996 until recently...

Comment (pcrespov) [reply Makefile]: what can i say ... I like the classics :-)

**Pull request merged**